### PR TITLE
Fix problem with runaway observation preloads

### LINKED ISSUE #449: Thundering Herd Of Duplicate (?) Observations Queries on Startup
Starting the app with `LOG_LEVEL=trace` nets a lot of chatter.

<details>

<code>
 INFO  [2023-10-16T14:56:43.106Z] TRACE prefetching
 INFO  [2023-10-16T14:56:43.107Z] DEBUG fetchNACObservations {
  "center": "SNFAC",
  "startDate": "2023-10-02",
  "endDate": "2023-10-16"
} query: ["nac-observations",{"host":"https://api.avalanche.org","center_id":"SNFAC","end_time":"latest"}]
 INFO  [2023-10-16T14:56:43.107Z] TRACE finished prefetching duration: "0 seconds" query: ["nac-observations",{"host":"https://api.avalanche.org","center_id":"SNFAC","end_time":"latest"}]
 INFO  [2023-10-16T14:56:43.107Z] DEBUG sending request uri: "POST https://api.avalanche.org/obs/v1/public/graphql data: {\"query\":\"\\n    query observations($center: String!, $startDate: String!, $endDate: String!) {\\n  getObservation...\",\"variables\":{\"center\":\"SNFAC\",\"startDate\":\"2023-10-02\",\"endDate\":\"2023-10-16\"}}"
 INFO  [2023-10-16T14:56:43.107Z] TRACE sending request data: "{\"query\":\"\\n    query observations($center: String!, $startDate: String!, $endDate: String!) {\\n  getObservation...\",\"variables\":{\"center\":\"SNFAC\",\"startDate\":\"2023-10-02\",\"endDate\":\"2023-10-16\"}}" uri: "POST https://api.avalanche.org/obs/v1/public/graphql data: {\"query\":\"\\n    query observations($center: String!, $startDate: String!, $endDate: String!) {\\n  getObservation...\",\"variables\":{\"center\":\"SNFAC\",\"startDate\":\"2023-10-02\",\"endDate\":\"2023-10-16\"}}"
 INFO  [2023-10-16T14:56:43.363Z] DEBUG received request response status: 200 uri: "POST https://api.avalanche.org/obs/v1/public/graphql data: \"{\\\"query\\\":\\\"\\\\n    query observations($center: String!, $startDate: String!, $endDate: String!) {\\\\n  ge...\""
 INFO  [2023-10-16T14:56:43.365Z] TRACE received request response data: "{\"data\":{\"getObservationList\":[]}}" status: 200 uri: "POST https://api.avalanche.org/obs/v1/public/graphql data: \"{\\\"query\\\":\\\"\\\\n    query observations($center: String!, $startDate: String!, $endDate: String!) {\\\\n  ge...\""
 INFO  [2023-10-16T14:56:43.369Z] TRACE prefetching
 INFO  [2023-10-16T14:56:43.370Z] DEBUG fetchNACObservations {
  "center": "SNFAC",
  "startDate": "2023-10-02",
  "endDate": "2023-10-16"
} query: ["nac-observations",{"host":"https://api.avalanche.org","center_id":"SNFAC","end_time":"latest"}]
 INFO  [2023-10-16T14:56:43.372Z] TRACE finished prefetching duration: "0 seconds" query: ["nac-observations",{"host":"https://api.avalanche.org","center_id":"SNFAC","end_time":"latest"}]
 INFO  [2023-10-16T14:56:43.373Z] DEBUG sending request uri: "POST https://api.avalanche.org/obs/v1/public/graphql data: {\"query\":\"\\n    query observations($center: String!, $startDate: String!, $endDate: String!) {\\n  getObservation...\",\"variables\":{\"center\":\"SNFAC\",\"startDate\":\"2023-10-02\",\"endDate\":\"2023-10-16\"}}"
 INFO  [2023-10-16T14:56:43.374Z] TRACE sending request data: "{\"query\":\"\\n    query observations($center: String!, $startDate: String!, $endDate: String!) {\\n  getObservation...\",\"variables\":{\"center\":\"SNFAC\",\"startDate\":\"2023-10-02\",\"endDate\":\"2023-10-16\"}}" uri: "POST https://api.avalanche.org/obs/v1/public/graphql data: {\"query\":\"\\n    query observations($center: String!, $startDate: String!, $endDate: String!) {\\n  getObservation...\",\"variables\":{\"center\":\"SNFAC\",\"startDate\":\"2023-10-02\",\"endDate\":\"2023-10-16\"}}"
 INFO  [2023-10-16T14:56:43.641Z] DEBUG received request response status: 200 uri: "POST https://api.avalanche.org/obs/v1/public/graphql data: \"{\\\"query\\\":\\\"\\\\n    query observations($center: String!, $startDate: String!, $endDate: String!) {\\\\n  ge...\""
 INFO  [2023-10-16T14:56:43.644Z] TRACE received request response data: "{\"data\":{\"getObservationList\":[]}}" status: 200 uri: "POST https://api.avalanche.org/obs/v1/public/graphql data: \"{\\\"query\\\":\\\"\\\\n    query observations($center: String!, $startDate: String!, $endDate: String!) {\\\\n  ge...\""
 INFO  [2023-10-16T14:56:43.649Z] TRACE prefetching
 INFO  [2023-10-16T14:56:43.650Z] DEBUG fetchNACObservations {
  "center": "SNFAC",
  "startDate": "2023-10-02",
  "endDate": "2023-10-16"
} query: ["nac-observations",{"host":"https://api.avalanche.org","center_id":"SNFAC","end_time":"latest"}]
 INFO  [2023-10-16T14:56:43.652Z] TRACE finished prefetching duration: "0 seconds" query: ["nac-observations",{"host":"https://api.avalanche.org","center_id":"SNFAC","end_time":"latest"}]
 INFO  [2023-10-16T14:56:43.653Z] DEBUG sending request uri: "POST https://api.avalanche.org/obs/v1/public/graphql data: {\"query\":\"\\n    query observations($center: String!, $startDate: String!, $endDate: String!) {\\n  getObservation...\",\"variables\":{\"center\":\"SNFAC\",\"startDate\":\"2023-10-02\",\"endDate\":\"2023-10-16\"}}"
 INFO  [2023-10-16T14:56:43.654Z] TRACE sending request data: "{\"query\":\"\\n    query observations($center: String!, $startDate: String!, $endDate: String!) {\\n  getObservation...\",\"variables\":{\"center\":\"SNFAC\",\"startDate\":\"2023-10-02\",\"endDate\":\"2023-10-16\"}}" uri: "POST https://api.avalanche.org/obs/v1/public/graphql data: {\"query\":\"\\n    query observations($center: String!, $startDate: String!, $endDate: String!) {\\n  getObservation...\",\"variables\":{\"center\":\"SNFAC\",\"startDate\":\"2023-10-02\",\"endDate\":\"2023-10-16\"}}"
 INFO  [2023-10-16T14:56:43.924Z] DEBUG received request response status: 200 uri: "POST https://api.avalanche.org/obs/v1/public/graphql data: \"{\\\"query\\\":\\\"\\\\n    query observations($center: String!, $startDate: String!, $endDate: String!) {\\\\n  ge...\""
 INFO  [2023-10-16T14:56:43.926Z] TRACE received request response data: "{\"data\":{\"getObservationList\":[]}}" status: 200 uri: "POST https://api.avalanche.org/obs/v1/public/graphql data: \"{\\\"query\\\":\\\"\\\\n    query observations($center: String!, $startDate: String!, $endDate: String!) {\\\\n  ge...\""
 INFO  [2023-10-16T14:56:43.931Z] TRACE prefetching
 INFO  [2023-10-16T14:56:43.932Z] DEBUG fetchNACObservations {
  "center": "SNFAC",
  "startDate": "2023-10-02",
  "endDate": "2023-10-16"
} query: ["nac-observations",{"host":"https://api.avalanche.org","center_id":"SNFAC","end_time":"latest"}]
 INFO  [2023-10-16T14:56:43.934Z] TRACE finished prefetching duration: "0 seconds" query: ["nac-observations",{"host":"https://api.avalanche.org","center_id":"SNFAC","end_time":"latest"}]
 INFO  [2023-10-16T14:56:43.935Z] DEBUG sending request uri: "POST https://api.avalanche.org/obs/v1/public/graphql data: {\"query\":\"\\n    query observations($center: String!, $startDate: String!, $endDate: String!) {\\n  getObservation...\",\"variables\":{\"center\":\"SNFAC\",\"startDate\":\"2023-10-02\",\"endDate\":\"2023-10-16\"}}"
 INFO  [2023-10-16T14:56:43.936Z] TRACE sending request data: "{\"query\":\"\\n    query observations($center: String!, $startDate: String!, $endDate: String!) {\\n  getObservation...\",\"variables\":{\"center\":\"SNFAC\",\"startDate\":\"2023-10-02\",\"endDate\":\"2023-10-16\"}}" uri: "POST https://api.avalanche.org/obs/v1/public/graphql data: {\"query\":\"\\n    query observations($center: String!, $startDate: String!, $endDate: String!) {\\n  getObservation...\",\"variables\":{\"center\":\"SNFAC\",\"startDate\":\"2023-10-02\",\"endDate\":\"2023-10-16\"}}"
 INFO  [2023-10-16T14:56:44.200Z] DEBUG received request response status: 200 uri: "POST https://api.avalanche.org/obs/v1/public/graphql data: \"{\\\"query\\\":\\\"\\\\n    query observations($center: String!, $startDate: String!, $endDate: String!) {\\\\n  ge...\""
 INFO  [2023-10-16T14:56:44.201Z] TRACE received request response data: "{\"data\":{\"getObservationList\":[]}}" status: 200 uri: "POST https://api.avalanche.org/obs/v1/public/graphql data: \"{\\\"query\\\":\\\"\\\\n    query observations($center: String!, $startDate: String!, $endDate: String!) {\\\\n  ge...\""
 INFO  [2023-10-16T14:56:44.201Z] TRACE prefetching
 INFO  [2023-10-16T14:56:44.202Z] DEBUG fetchNACObservations {
  "center": "SNFAC",
  "startDate": "2023-10-02",
  "endDate": "2023-10-16"
} query: ["nac-observations",{"host":"https://api.avalanche.org","center_id":"SNFAC","end_time":"latest"}]
 INFO  [2023-10-16T14:56:44.202Z] TRACE finished prefetching duration: "0 seconds" query: ["nac-observations",{"host":"https://api.avalanche.org","center_id":"SNFAC","end_time":"latest"}]
 INFO  [2023-10-16T14:56:44.202Z] DEBUG sending request uri: "POST https://api.avalanche.org/obs/v1/public/graphql data: {\"query\":\"\\n    query observations($center: String!, $startDate: String!, $endDate: String!) {\\n  getObservation...\",\"variables\":{\"center\":\"SNFAC\",\"startDate\":\"2023-10-02\",\"endDate\":\"2023-10-16\"}}"
 INFO  [2023-10-16T14:56:44.202Z] TRACE sending request data: "{\"query\":\"\\n    query observations($center: String!, $startDate: String!, $endDate: String!) {\\n  getObservation...\",\"variables\":{\"center\":\"SNFAC\",\"startDate\":\"2023-10-02\",\"endDate\":\"2023-10-16\"}}" uri: "POST https://api.avalanche.org/obs/v1/public/graphql data: {\"query\":\"\\n    query observations($center: String!, $startDate: String!, $endDate: String!) {\\n  getObservation...\",\"variables\":{\"center\":\"SNFAC\",\"startDate\":\"2023-10-02\",\"endDate\":\"2023-10-16\"}}"
 INFO  [2023-10-16T14:56:44.456Z] DEBUG received request response status: 200 uri: "POST https://api.avalanche.org/obs/v1/public/graphql data: \"{\\\"query\\\":\\\"\\\\n    query observations($center: String!, $startDate: String!, $endDate: String!) {\\\\n  ge...\""
 INFO  [2023-10-16T14:56:44.458Z] TRACE received request response data: "{\"data\":{\"getObservationList\":[]}}" status: 200 uri: "POST https://api.avalanche.org/obs/v1/public/graphql data: \"{\\\"query\\\":\\\"\\\\n    query observations($center: String!, $startDate: String!, $endDate: String!) {\\\\n  ge...\""
 INFO  [2023-10-16T14:56:44.463Z] TRACE prefetching
 INFO  [2023-10-16T14:56:44.465Z] DEBUG fetchNACObservations {
  "center": "SNFAC",
  "startDate": "2023-10-02",
  "endDate": "2023-10-16"
} query: ["nac-observations",{"host":"https://api.avalanche.org","center_id":"SNFAC","end_time":"latest"}]
 INFO  [2023-10-16T14:56:44.466Z] TRACE finished prefetching duration: "0 seconds" query: ["nac-observations",{"host":"https://api.avalanche.org","center_id":"SNFAC","end_time":"latest"}]
 INFO  [2023-10-16T14:56:44.468Z] DEBUG sending request uri: "POST https://api.avalanche.org/obs/v1/public/graphql data: {\"query\":\"\\n    query observations($center: String!, $startDate: String!, $endDate: String!) {\\n  getObservation...\",\"variables\":{\"center\":\"SNFAC\",\"startDate\":\"2023-10-02\",\"endDate\":\"2023-10-16\"}}"
 INFO  [2023-10-16T14:56:44.468Z] TRACE sending request data: "{\"query\":\"\\n    query observations($center: String!, $startDate: String!, $endDate: String!) {\\n  getObservation...\",\"variables\":{\"center\":\"SNFAC\",\"startDate\":\"2023-10-02\",\"endDate\":\"2023-10-16\"}}" uri: "POST https://api.avalanche.org/obs/v1/public/graphql data: {\"query\":\"\\n    query observations($center: String!, $startDate: String!, $endDate: String!) {\\n  getObservation...\",\"variables\":{\"center\":\"SNFAC\",\"startDate\":\"2023-10-02\",\"endDate\":\"2023-10-16\"}}"
 INFO  [2023-10-16T14:56:44.735Z] DEBUG received request response status: 200 uri: "POST https://api.avalanche.org/obs/v1/public/graphql data: \"{\\\"query\\\":\\\"\\\\n    query observations($center: String!, $startDate: String!, $endDate: String!) {\\\\n  ge...\""
 INFO  [2023-10-16T14:56:44.737Z] TRACE received request response data: "{\"data\":{\"getObservationList\":[]}}" status: 200 uri: "POST https://api.avalanche.org/obs/v1/public/graphql data: \"{\\\"query\\\":\\\"\\\\n    query observations($center: String!, $startDate: String!, $endDate: String!) {\\\\n  ge...\""
 INFO  [2023-10-16T14:56:44.743Z] TRACE prefetching
 INFO  [2023-10-16T14:56:44.745Z] DEBUG fetchNACObservations {
  "center": "SNFAC",
  "startDate": "2023-10-02",
  "endDate": "2023-10-16"
} query: ["nac-observations",{"host":"https://api.avalanche.org","center_id":"SNFAC","end_time":"latest"}]
 INFO  [2023-10-16T14:56:44.746Z] TRACE finished prefetching duration: "0 seconds" query: ["nac-observations",{"host":"https://api.avalanche.org","center_id":"SNFAC","end_time":"latest"}]
 INFO  [2023-10-16T14:56:44.748Z] DEBUG sending request uri: "POST https://api.avalanche.org/obs/v1/public/graphql data: {\"query\":\"\\n    query observations($center: String!, $startDate: String!, $endDate: String!) {\\n  getObservation...\",\"variables\":{\"center\":\"SNFAC\",\"startDate\":\"2023-10-02\",\"endDate\":\"2023-10-16\"}}"
 INFO  [2023-10-16T14:56:44.749Z] TRACE sending request data: "{\"query\":\"\\n    query observations($center: String!, $startDate: String!, $endDate: String!) {\\n  getObservation...\",\"variables\":{\"center\":\"SNFAC\",\"startDate\":\"2023-10-02\",\"endDate\":\"2023-10-16\"}}" uri: "POST https://api.avalanche.org/obs/v1/public/graphql data: {\"query\":\"\\n    query observations($center: String!, $startDate: String!, $endDate: String!) {\\n  getObservation...\",\"variables\":{\"center\":\"SNFAC\",\"startDate\":\"2023-10-02\",\"endDate\":\"2023-10-16\"}}"
 INFO  [2023-10-16T14:56:45.010Z] DEBUG received request response status: 200 uri: "POST https://api.avalanche.org/obs/v1/public/graphql data: \"{\\\"query\\\":\\\"\\\\n    query observations($center: String!, $startDate: String!, $endDate: String!) {\\\\n  ge...\""
 INFO  [2023-10-16T14:56:45.012Z] TRACE received request response data: "{\"data\":{\"getObservationList\":[]}}" status: 200 uri: "POST https://api.avalanche.org/obs/v1/public/graphql data: \"{\\\"query\\\":\\\"\\\\n    query observations($center: String!, $startDate: String!, $endDate: String!) {\\\\n  ge...\""
 INFO  [2023-10-16T14:56:45.017Z] TRACE prefetching
 INFO  [2023-10-16T14:56:45.019Z] DEBUG fetchNACObservations {
  "center": "SNFAC",
  "startDate": "2023-10-02",
  "endDate": "2023-10-16"
} query: ["nac-observations",{"host":"https://api.avalanche.org","center_id":"SNFAC","end_time":"latest"}]
 INFO  [2023-10-16T14:56:45.020Z] TRACE finished prefetching duration: "0 seconds" query: ["nac-observations",{"host":"https://api.avalanche.org","center_id":"SNFAC","end_time":"latest"}]
 INFO  [2023-10-16T14:56:45.022Z] DEBUG sending request uri: "POST https://api.avalanche.org/obs/v1/public/graphql data: {\"query\":\"\\n    query observations($center: String!, $startDate: String!, $endDate: String!) {\\n  getObservation...\",\"variables\":{\"center\":\"SNFAC\",\"startDate\":\"2023-10-02\",\"endDate\":\"2023-10-16\"}}"
 INFO  [2023-10-16T14:56:45.022Z] TRACE sending request data: "{\"query\":\"\\n    query observations($center: String!, $startDate: String!, $endDate: String!) {\\n  getObservation...\",\"variables\":{\"center\":\"SNFAC\",\"startDate\":\"2023-10-02\",\"endDate\":\"2023-10-16\"}}" uri: "POST https://api.avalanche.org/obs/v1/public/graphql data: {\"query\":\"\\n    query observations($center: String!, $startDate: String!, $endDate: String!) {\\n  getObservation...\",\"variables\":{\"center\":\"SNFAC\",\"startDate\":\"2023-10-02\",\"endDate\":\"2023-10-16\"}}"
 INFO  [2023-10-16T14:56:45.299Z] DEBUG received request response status: 200 uri: "POST https://api.avalanche.org/obs/v1/public/graphql data: \"{\\\"query\\\":\\\"\\\\n    query observations($center: String!, $startDate: String!, $endDate: String!) {\\\\n  ge...\""
 INFO  [2023-10-16T14:56:45.301Z] TRACE received request response data: "{\"data\":{\"getObservationList\":[]}}" status: 200 uri: "POST https://api.avalanche.org/obs/v1/public/graphql data: \"{\\\"query\\\":\\\"\\\\n    query observations($center: String!, $startDate: String!, $endDate: String!) {\\\\n  ge...\""
 INFO  [2023-10-16T14:56:45.307Z] TRACE prefetching
 INFO  [2023-10-16T14:56:45.308Z] DEBUG fetchNACObservations {
  "center": "SNFAC",
  "startDate": "2023-10-02",
  "endDate": "2023-10-16"
} query: ["nac-observations",{"host":"https://api.avalanche.org","center_id":"SNFAC","end_time":"latest"}]
</code>

</details>

### PR DESCRIPTION
**Pull request by floatplane**
This is straight up voodoo. The diff has my best guess about the root cause, but I googled around for `prefetchInfiniteQuery POST` and various other queries without finding anything referring to this.

Fixes #449

### CODE REVIEW

**Comment by floatplane** [comment]
@stevekuznetsov not sure if this gives you a new angle that you want to investigate further? otherwise I probably would be in favor of merging somewhat quickly as this is a drag on perf (and a drag on the NAC servers)

**Comment by stevekuznetsov** [comment]
What was the diff that actually caused the issue? Maybe we can capture that the issue & pick at that later, but if this stops the bleeding let's get it in

**Comment by floatplane** [comment]
> What was the diff that actually caused the issue? Maybe we can capture that the issue & pick at that later, but if this stops the bleeding let's get it in

My theory was that it started when we transitioned to GraphQL queries (which require `POST` instead of `GET`) but I didn't go back in time to verify it! That said, I don't see how tanstack could get in a weird state just because of that, because tanstack has no idea whether you're posting or getting - it just calls your `queryFn`.

Is there something returned in the response to the GQL query that signals "don't cache this by default" to tanstack? 🤷 

tl;dr I have no idea what is going on here, this is still a mystery to me 😞

**Comment by stevekuznetsov** [comment]
Wild. I usually run with high debug log-level and SNFAC and did not see it until I made the issue so it might have been more recent. I guess we can invest more time later if it comes back to bite us somehow.
